### PR TITLE
enhance: always check parent directory (fix #8679)

### LINKED ISSUE #8679: [bug] tauri resource directory is hard-coded
### Describe the bug

Tauri resource directory location is hard-coded making Tauri fail when it can't read or write from it, this is especially a problem in immutable distros like NixOS.

### Reproduction

https://github.com/NixOS/nixpkgs/pull/265771 - build this without the tauri patch in it.

### Expected behavior

For it to succeed in building.

### Full `tauri info` output

```text
N/A
```

### Stack trace

_No response_

### Additional context

There is a fix in the Github pr I linked in the files it's called [tauri-env-resource-dir.patch](https://github.com/NixOS/nixpkgs/blob/728e957b03d9194b7d277907cb1bc154081cf35a/pkgs/by-name/do/dorion/tauri-env-resource-dir.patch)

### PR DESCRIPTION
**Pull request by griffi-gh**
closes #8679

### CODE REVIEW

File: crates/tauri-utils/src/platform.rs [hunk 1]
```diff
@@ -309,12 +309,13 @@ fn resource_dir_from<P: AsRef<std::path::Path>>(
 
   #[cfg(target_os = "linux")]
   {
-    res = if curr_dir.ends_with("/data/usr/bin") {
+    // (canonicalize checks for existence, so there's no need for an extra check)
+    res = if let Ok(bundle_dir) = exe_dir
+      .join(format!("../lib/{}", package_info.name))
+      .canonicalize()
+    {
       // running from the deb bundle dir
```
Comment (lucasfernog) [inline crates/tauri-utils/src/platform.rs]:
```suggestion
```

**Review by lucasfernog — DISMISSED** [review]
this should be fine but i'd also like input from @tweidinger since we're no longer checking the path we're at this allows the binary/resources to be anywhere in the filesystem - as long as the resources are in ../lib/name

**Review by tweidinger — DISMISSED** [review]
I think it is fine to not ensure the path ends with the `/data/usr/bin` suffix, since the initial path is derived from the executable path and I don't think we make security specific promises of the resource dir.

**Review by lucasfernog — APPROVED** [review]

**Comment by absassi** [comment]
Thanks for the fix.
Would it be possible to backport this to V1 as well? At the moment, I can't upgrade to V2 due to #9039.

**Comment by FabianLars** [comment]
Sure, it's possible and we still accept prs for v1 :)

**Comment by MarielaSassi** [comment]
> Sure, it's possible and we still accept prs for v1 :)

Backport done, please have a look. @FabianLars 
https://github.com/tauri-apps/tauri/pull/11960

**Pull request merged**